### ISSUE TITLE
We created multiple workspaces for user in onboarding

### ISSUE THREAD

**Issue by danielrvidal** [Expensify/App#52519]
This is being discussed here so I'm just creating a tracking issue for us to get to the bottom of it with @marcaaron @trjExpensify 

https://expensify.slack.com/archives/C07HPDRELLD/p1731537294677189

**Comment by marcaaron** [comment]
I think @carlosmiceli is working on moving the policy creation to the `CreateAccount` step. I am wondering if this is at all related to those changes. At least, it seems like it will be fixed by them since we will no longer be creating the policy via an API call triggered in App.

**Comment by trjExpensify** [comment]
Interesting. Carlos, how does that work if you create an account on new.expensify.com directly where there's no signUpQualifiers to choose? 🤔

**Comment by carlosmiceli** [comment]
What @marcaaron says is correct, that's what I'm working on. @trjExpensify so far, we would just not create a workspace for any signup that hasn't selected a qualifier. In fact, creating an account will _not_ fail if for some reason the workspace fails to get created. But let's check with @anmurali to see if we'd like to add/change some logic about it.

**Comment by trjExpensify** [comment]
>@trjExpensify so far, we would just not create a workspace for any signup that hasn't selected a qualifier

Okay, which means they would then see the onboarding modal and selecting the manage my team path would create a workspace?

**Comment by trjExpensify** [comment]
Sorry, maybe I'm not explaining myself well. 

1. You said you're working on creating a workspace after you choose this on the expensify.com home page:

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/13383a1c-3c68-4c0a-aa62-a4b0c2da3873">


2. Today, the timing of creating the workspace is after you click this: 

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/4e897b7f-f69e-48f1-8e86-0759e4f982a1">

Q: In the expensify.com sign-up flow, how are you avoiding creating two workspaces on sign-up? Are you skipping that "How many employees do you have?" screen, and as a result, not triggering the `CreateWorkspace` command a second time? 

Q: If you sign-up via new.expensify.com where there are no signUpQualifiers to select on the home page, will you still see the "How many employees do you have?" screen if you choose "Manage my team's expenses" in the first step, and thus still trigger `CreateWorkspace` at that point like today? 

<img width="1507" alt="image" src="https://github.com/user-attachments/assets/91619720-f2d5-473d-9108-6bfff839b1e4">

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/f0d03813-eecf-4898-87ba-1f9995a79196">

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/03e74222-ff08-41ea-bbcd-7533d2f47f74">

**Comment by carlosmiceli** [comment]
> In the expensify.com sign-up flow, how are you avoiding creating two workspaces on sign-up? Are you skipping that "How many employees do you have?" screen, and as a result, not triggering the CreateWorkspace command a second time?

I haven't got to this point yet, but I think as Marc said, we're going to remove the current CreateWorkspace logic of creating the policy in App during onboarding (at least temporarily so we don't create duplicates).

> If you sign-up via new.expensify.com where there are no signUpQualifiers to select on the home page, will you still see the "How many employees do you have?" screen if you choose "Manage my team's expenses" in the first step, and thus still trigger CreateWorkspace at that point like today?

To clarify, I'm not working in anything that would change any logic that happens after/during signup through new.expensify.com, other than replacing the logic as I mentioned above. You will see whatever you see while signing up now. This means that no workspace will get created when signing up through new.expensify.com until we actually want to implement some logic around that.

I hope I'm understanding the question 😅

**Comment by carlosmiceli** [comment]
Ah, gotcha! Yup, that sounds good, we can do that. By the way, while it's an edge case, we should consider [this](https://expensify.slack.com/archives/C07HPDRELLD/p1730372974697939?thread_ts=1730238013.621219&cid=C07HPDRELLD) again then. I think with this new situation, it's justified to just hide the back button instead of having to implement more complicated logic to skip the workspace creation if they happen to go back.

**Comment by trjExpensify** [comment]
Hm, but isn't that logic going to exist anyway? As in, we won't show the employee count selector screen if they've chosen a signUpQualifier on sign-up? So regardless of if they go back and forward again, the page that triggers the CreateWorkspace command won't show to them.

**Comment by trjExpensify** [comment]
>so I interpreted that clicking Continue in that screen is what triggered CreateWorkspace.

Correct, that's my understanding as well. But if the page is skipped, meaning it doesn't show at all for this use-case, then going back/forward doesn't matter does it? They'll never see that "employee count" page and therefore no Continue button to trigger creating a workspace. So it's not clear to me why we need to restrict going back..

**Comment by carlosmiceli** [comment]
Right, what I'm saying is that, currently, after a user signups and skips those screens and sees the accounting software screen in the onboarding modal, they can go back clicking on the top left of the modal and go back to the employee count. It's not skipped in terms of being hidden, but just being taken straight to the software sections; all previous sections can still be seen if navigating back.

**Comment by trjExpensify** [comment]
Couldn't we hide it instead though? 🤔

**Comment by carlosmiceli** [comment]
Probably, but not sure how to do that since I haven't worked on that logic, that's probably a good question for someone with more knowledge there. Maybe @tgolen has some advice here? I still think hiding the back button is way easier, though 😅

**Comment by carlosmiceli** [comment]
Actually, thinking deeper about it, it's more future proof to hide the screens as you suggest, in case we keep expanding the onboarding steps.

**Comment by danielrvidal** [comment]
Not overdue, this is getting worked on daily.

**Comment by danielrvidal** [comment]
Still working through the updates in slack.

**Comment by carlosmiceli** [comment]
Will create issue today to align the ND fix with the BE work for https://github.com/Expensify/Expensify/issues/437845

**Comment by carlosmiceli** [comment]
Created here: https://github.com/Expensify/App/issues/53326

**Comment by danielrvidal** [comment]
This is being worked on!

**Comment by carlosmiceli** [comment]
Still working on this on many fronts 😅

**Comment by danielrvidal** [comment]
Still being worked on, I'm going to move it to weekly since it's just a tracking issue for more urgent issues.

**Comment by danielrvidal** [comment]
The Front PR is in review so we're still making progress!

**Comment by danielrvidal** [comment]
@carlosmiceli where are we at with this one? Still waiting on this to be done?
https://github.com/Expensify/App/issues/53326

**Comment by carlosmiceli** [comment]
The contributor was instructed of the changes to be implemented yesterday, while I'm pushing the last Web-E review suggestion today and we can have the BE ready to merge.

**Comment by danielrvidal** [comment]
Any updates on whats going on with this one? Not sure if anyone is still making progress or if it will come next week?

**Comment by carlosmiceli** [comment]
I asked a few days ago for pullerbear to bring more reviewers to both BE PRs because the reviews have been very slow since @marcaaron went on paternity leave, as he was the reviewer giving consistent reviews. I'm paying attention to it daily but I guess many people are OOO these days too...

**Comment by carlosmiceli** [comment]
Having said that, if I get a new response from the other reviewers, I'll make the push/pitch to just merge since it's been confirmed to work and we can hopefully have it done either this or the next.

**Comment by carlosmiceli** [comment]
Auth PR was merged yesterday. Once it's deployed, Web-E will be merged as well and this will be complete on the BE side. This recent PR in the FE should complete the cleanup: https://github.com/Expensify/App/pull/55168

I'll make sure to test everything once we merge Web-E, to confirm it's working properly. We should be done this week.

**Comment by danielrvidal** [comment]
@carlosmiceli do we still need this one to be open?

**Comment by carlosmiceli** [comment]
Nope! Closing.
